keycloakClient/keycloakClient.go

```diff
@@ -79,6 +85,28 @@ func New(config Config) (*Client, error) {
 	return &client, nil
 }
 
+// AutoRefreshToken starts a process where an access token is kept perpetually
+// warm in the cache, refreshing itself five seconds before it expires.
+func (c *Client) AutoRefreshToken(realm string, username string, password string, onFailure func(error)) {
+	info, err := c.GetTokenInfo(realm, username, password, true)
+	if err != nil {
+		// Unable to fetch the token, allow userland to determine the correct
+		// behavior here -- retry, panic, log, etc...
+		onFailure(err)
+		return
+	}
+	// Refresh 5 seconds before the auth token expires
+	nextRefresh := info.Expires.Sub(time.Now().Add(5 * time.Second))
```

Comment:
you can ignore me if its too annoying, but can we have this come from the client config, default it to 5 if it's not set, and have identity service when setting up the client use a value from an environment variable to set it for our uses